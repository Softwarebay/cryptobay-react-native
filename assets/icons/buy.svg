<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_4028_1970)">
<path d="M19.1119 6.49996C18.8816 6.19602 18.6318 5.90741 18.3639 5.63596C17.5282 4.80022 16.5361 4.13727 15.4441 3.68497C14.3522 3.23267 13.1818 2.99988 11.9999 2.99988C10.818 2.99988 9.6477 3.23267 8.55576 3.68497C7.46383 4.13727 6.47167 4.80022 5.63594 5.63596C5.36839 5.90766 5.11854 6.19626 4.88794 6.49996" stroke="#FEBF32" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M4.88794 17.5C5.11794 17.799 5.36194 18.09 5.63594 18.364C6.47167 19.1997 7.46383 19.8627 8.55576 20.315C9.6477 20.7673 10.818 21.0001 11.9999 21.0001C13.1818 21.0001 14.3522 20.7673 15.4441 20.315C16.5361 19.8627 17.5282 19.1997 18.3639 18.364C18.6379 18.09 18.8809 17.799 19.1119 17.5" stroke="#FEBF32" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M6.521 14.571H3.5V12.04H6.519C7.218 12.04 7.785 12.607 7.785 13.306C7.78513 13.4721 7.75253 13.6366 7.68907 13.7901C7.62561 13.9435 7.53252 14.083 7.41514 14.2005C7.29775 14.3179 7.15836 14.4111 7.00494 14.4747C6.85152 14.5383 6.68707 14.571 6.521 14.571V14.571Z" stroke="#FEBF32" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M3.5 12.04V9.42896H6.2C6.54637 9.42896 6.87856 9.56655 7.12348 9.81147C7.3684 10.0564 7.506 10.3886 7.506 10.735C7.506 11.0813 7.3684 11.4135 7.12348 11.6584C6.87856 11.9034 6.54637 12.041 6.2 12.041" stroke="#FEBF32" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M17.2141 9.42896L19.3571 12.344L21.5001 9.42896" stroke="#FEBF32" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M19.3601 14.53V12.34" stroke="#FEBF32" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M14.6429 9.42896V12.429C14.6429 12.9973 14.4172 13.5424 14.0153 13.9443C13.6134 14.3462 13.0683 14.572 12.4999 14.572C11.9316 14.572 11.3865 14.3462 10.9846 13.9443C10.5827 13.5424 10.3569 12.9973 10.3569 12.429V9.42896" stroke="#FEBF32" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</g>
<defs>
<clipPath id="clip0_4028_1970">
<rect width="24" height="24" fill="white"/>
</clipPath>
</defs>
</svg>
