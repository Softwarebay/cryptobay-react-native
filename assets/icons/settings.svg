<svg width="25" height="24" viewBox="0 0 25 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M14.41 10.09C14.9165 10.5966 15.2011 11.2836 15.2011 12C15.2011 12.7164 14.9165 13.4034 14.41 13.91C13.9034 14.4166 13.2164 14.7011 12.5 14.7011C11.7836 14.7011 11.0965 14.4166 10.59 13.91C10.0834 13.4034 9.79883 12.7164 9.79883 12C9.79883 11.6453 9.8687 11.294 10.0044 10.9663C10.1402 10.6386 10.3392 10.3408 10.59 10.09C10.8408 9.83917 11.1386 9.64021 11.4663 9.50446C11.794 9.36872 12.1453 9.29885 12.5 9.29885C13.2164 9.29885 13.9034 9.58344 14.41 10.09" stroke="#ABAFC4" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M5.74993 12C5.74993 12.297 5.77693 12.594 5.81293 12.882L4.22493 14.124C4.05461 14.2574 3.93817 14.4478 3.89702 14.6602C3.85586 14.8725 3.89276 15.0926 4.00093 15.28L5.41293 17.723C5.52107 17.9101 5.69304 18.0519 5.89729 18.1224C6.10155 18.193 6.32439 18.1875 6.52493 18.107L7.94693 17.536C8.08536 17.4824 8.23477 17.4634 8.3822 17.4807C8.52963 17.4979 8.67062 17.5509 8.79293 17.635C9.01293 17.781 9.24093 17.915 9.47693 18.035C9.74693 18.172 9.94293 18.417 9.98593 18.717L10.2029 20.23C10.2659 20.672 10.6449 21 11.0909 21H13.9079C14.1238 21 14.3325 20.9221 14.4956 20.7807C14.6587 20.6393 14.7654 20.4437 14.7959 20.23L15.0129 18.718C15.0366 18.5712 15.0955 18.4323 15.1847 18.3134C15.2739 18.1944 15.3907 18.0989 15.5249 18.035C15.7599 17.917 15.9869 17.784 16.2059 17.639C16.3286 17.554 16.4701 17.5002 16.6183 17.4825C16.7664 17.4647 16.9167 17.4834 17.0559 17.537L18.4749 18.107C18.6755 18.1873 18.8983 18.1927 19.1025 18.1221C19.3067 18.0516 19.4786 17.9099 19.5869 17.723L20.9989 15.28C21.1071 15.0926 21.144 14.8725 21.1028 14.6602C21.0617 14.4478 20.9453 14.2574 20.7749 14.124L19.1869 12.882C19.2229 12.594 19.2499 12.297 19.2499 12C19.2499 11.703 19.2229 11.406 19.1869 11.118L20.7749 9.876C20.9453 9.74261 21.0617 9.55222 21.1028 9.33984C21.144 9.12745 21.1071 8.90735 20.9989 8.72L19.5869 6.277C19.4788 6.08991 19.3068 5.94809 19.1026 5.87755C18.8983 5.80702 18.6755 5.8125 18.4749 5.893L17.0559 6.463C16.9166 6.51634 16.7664 6.53492 16.6183 6.51715C16.4702 6.49938 16.3287 6.44578 16.2059 6.361C15.987 6.21555 15.7596 6.08332 15.5249 5.965C15.3907 5.90113 15.2739 5.8056 15.1847 5.68663C15.0955 5.56766 15.0366 5.4288 15.0129 5.282L14.7969 3.77C14.7664 3.55627 14.6597 3.36074 14.4966 3.2193C14.3335 3.07785 14.1248 2.99999 13.9089 3H11.0919C10.876 2.99999 10.6674 3.07785 10.5042 3.2193C10.3411 3.36074 10.2345 3.55627 10.2039 3.77L9.98593 5.284C9.96228 5.43026 9.90365 5.56865 9.81503 5.68739C9.72641 5.80613 9.61042 5.90171 9.47693 5.966C9.24093 6.086 9.01293 6.221 8.79293 6.366C8.67024 6.44971 8.52906 6.50236 8.38152 6.51943C8.23397 6.5365 8.08449 6.51748 7.94593 6.464L6.52493 5.893C6.32439 5.8125 6.10155 5.80702 5.89729 5.87755C5.69304 5.94809 5.52107 6.08991 5.41293 6.277L4.00093 8.72C3.89276 8.90735 3.85586 9.12745 3.89702 9.33984C3.93817 9.55222 4.05461 9.74261 4.22493 9.876L5.81293 11.118C5.77372 11.4104 5.75268 11.705 5.74993 12V12Z" stroke="#ABAFC4" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
