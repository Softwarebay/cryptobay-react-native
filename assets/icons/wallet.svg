<svg width="25" height="24" viewBox="0 0 25 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M17.0949 13.125C16.8879 13.126 16.7209 13.294 16.7209 13.501C16.7209 13.708 16.8889 13.876 17.0959 13.875C17.3029 13.875 17.4709 13.707 17.4709 13.5C17.4709 13.293 17.3029 13.125 17.0949 13.125" stroke="#FEBF32" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M3.5 5V18.5C3.5 19.605 4.395 20.5 5.5 20.5H19.5C20.605 20.5 21.5 19.605 21.5 18.5V8.5C21.5 7.395 20.605 6.5 19.5 6.5H5C4.172 6.5 3.5 5.828 3.5 5V5C3.5 4.172 4.172 3.5 5 3.5H17.5" stroke="#FEBF32" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
