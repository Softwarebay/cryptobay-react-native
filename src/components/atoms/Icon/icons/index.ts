export { ArrowIcon } from './components/ArrowIcon';
export { BuyIcon } from './components/BuyIcon';
export { CheckIcon } from './components/CheckIcon';
export { ElipseProfileIcon } from './components/ElipseProfileIcon';
export { PlusIcon } from './components/PlusIcon';
export { SettingsIcon } from './components/SettingsIcon';
export { SwapIcon } from './components/SwapIcon';
export { TickArrowIcon } from './components/TickArrowIcon';
export { WalletIcon } from './components/WalletIcon';
