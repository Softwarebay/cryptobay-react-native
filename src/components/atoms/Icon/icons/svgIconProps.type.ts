export type SvgIconProps = {
  width: number;
  height: number;
  color?: string;
  testID: string;
};
