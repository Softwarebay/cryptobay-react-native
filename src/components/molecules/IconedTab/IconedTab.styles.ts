import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  iconedTab: {
    width: '33%',
    paddingTop: 10,
    paddingBottom: 34,
    justifyContent: 'center',
    alignItems: 'center',
  },
  checkedLabel: {
    color: '#FEBF32',
  },
  uncheckedLabel: {
    color: '#ABAFC4',
  },
});
