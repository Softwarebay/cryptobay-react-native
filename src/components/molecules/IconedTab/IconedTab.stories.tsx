import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';
import { Icons } from '../../atoms/Icon/icons.enum';

import { IconedTab } from './IconedTab';

// Styles to better story view
const storyStyles = {
  borderWidth: 1,
  borderColor: 'white',
};

storiesOf('Molecules/IconedTab', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Unchecked Iconed Label', () => (
    <IconedTab
      label={text('label', 'Wallet')}
      checked={false}
      iconType={Icons.WALLET}
      onPress={action('onPress')}
      style={storyStyles}
    />
  ))
  .add('Checked Iconed Label', () => (
    <IconedTab
      label={text('label', 'Wallet')}
      checked={true}
      iconType={Icons.WALLET}
      onPress={action('onPress')}
      style={storyStyles}
    />
  ));
