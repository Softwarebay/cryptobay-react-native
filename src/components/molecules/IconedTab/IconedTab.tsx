import React from 'react';
import { TouchableOpacity, ViewStyle } from 'react-native';

import { Label } from '../../atoms';
import { Icon } from '../../atoms/Icon/Icon';
import { Icons } from '../../atoms/Icon/icons.enum';

import { styles } from './IconedTab.styles';

type IconedTabProps = {
  label: string;
  checked: boolean;
  iconType: Icons;
  onPress: () => void;
  style?: ViewStyle | ViewStyle[];
  testID?: string;
  labelTestID?: string;
  iconTestID?: string;
};

export const IconedTab: React.FC<IconedTabProps> = ({
  label,
  checked,
  iconType,
  onPress,
  style,
  testID,
  labelTestID,
  iconTestID,
}: IconedTabProps) => {
  const iconColor = checked ? '#FEBF32' : '#ABAFC4';
  const handleLabelStylesByChecked = checked
    ? styles.checkedLabel
    : styles.uncheckedLabel;

  return (
    <TouchableOpacity
      style={[styles.iconedTab, style]}
      testID={testID}
      onPress={onPress}>
      <Icon type={iconType} color={iconColor} testID={iconTestID} />
      <Label
        value={label}
        variant="tertiary"
        style={handleLabelStylesByChecked}
        testID={labelTestID}
      />
    </TouchableOpacity>
  );
};
