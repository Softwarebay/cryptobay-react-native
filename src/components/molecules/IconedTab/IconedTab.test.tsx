import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { Icons } from '../../atoms/Icon/icons.enum';

import { IconedTab } from './IconedTab';

const mockOnPress = jest.fn();

describe('Iconed Tab Molecule', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <IconedTab
          label="Wallet"
          checked={false}
          iconType={Icons.WALLET}
          onPress={mockOnPress}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure iconed tab is checkable', () => {
    const { rerender } = render(
      <IconedTab
        label="Settings"
        checked={false}
        iconType={Icons.SETTINGS}
        onPress={mockOnPress}
        testID="molecule-iconed-tab"
        labelTestID="iconed-tab-label"
        iconTestID="iconed-tab-icon"
      />,
    );
    const label = screen.getByTestId('iconed-tab-label');
    expect(label).toHaveStyle({
      color: '#ABAFC4',
    });
    expect(label).toHaveTextContent('Settings');
    rerender(
      <IconedTab
        label="Swap"
        checked={true}
        iconType={Icons.SWAP}
        onPress={mockOnPress}
        testID="molecule-iconed-tab"
        labelTestID="iconed-tab-label"
        iconTestID="iconed-tab-icon"
      />,
    );
    expect(label).toHaveStyle({
      color: '#FEBF32',
    });
  });

  test('ensure icons renders correctly', () => {
    render(
      <IconedTab
        label="Sent"
        checked={false}
        iconType={Icons.ARROW}
        onPress={mockOnPress}
        testID="molecule-iconed-tab"
        labelTestID="iconed-tab-label"
        iconTestID="iconed-tab-icon"
      />,
    );
    expect(screen.queryByTestId('svg-icon-arrow')).not.toBeNull();
  });
});
