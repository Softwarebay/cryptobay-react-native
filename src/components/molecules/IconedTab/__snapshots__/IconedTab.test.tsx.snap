// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Iconed Tab Molecule Component Snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 34,
      "paddingTop": 10,
      "width": "33%",
    }
  }
>
  <View>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="svg-icon-wallet"
      vbHeight={24}
      vbWidth={25}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M17.0949 13.125C16.8879 13.126 16.7209 13.294 16.7209 13.501C16.7209 13.708 16.8889 13.876 17.0959 13.875C17.3029 13.875 17.4709 13.707 17.4709 13.5C17.4709 13.293 17.3029 13.125 17.0949 13.125"
          propList={
            Array [
              "stroke",
            ]
          }
          stroke={
            Object {
              "payload": 4289441732,
              "type": 0,
            }
          }
        />
        <RNSVGPath
          d="M3.5 5V18.5C3.5 19.605 4.395 20.5 5.5 20.5H19.5C20.605 20.5 21.5 19.605 21.5 18.5V8.5C21.5 7.395 20.605 6.5 19.5 6.5H5C4.172 6.5 3.5 5.828 3.5 5V5C3.5 4.172 4.172 3.5 5 3.5H17.5"
          propList={
            Array [
              "stroke",
            ]
          }
          stroke={
            Object {
              "payload": 4289441732,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins",
        },
        Object {
          "fontSize": 12,
          "fontWeight": "700",
          "lineHeight": 18,
        },
        Object {
          "color": "#ABAFC4",
        },
      ]
    }
  >
    Wallet
  </Text>
</View>
`;
